overlord/servicestate/servicestate.go

```diff
@@ -49,9 +49,42 @@ type Instruction struct {
 
 type ServiceActionConflictError struct{ error }
 
+type ResolvedAppInfo struct {
```

Comment:
I'm not a fan of this type, it avoids signature churn and taken alone looks elegant but at the cost of a lot of infrastructure for it in terms of helpers. Some duplicating functionality. Partly it's because go doesn't have generics, but that's how things are. I would prefer not to need it.